gst_all_1: 1.24.10 → 1.26.0

jtojnar opened the pull request:
https://gstreamer.freedesktop.org/releases/1.26/

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->
- [x] Skimmed the [release notes](https://gstreamer.freedesktop.org/releases/1.26/)
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jtojnar committed:
orc: 0.4.40 → 0.4.41

https://gitlab.freedesktop.org/gstreamer/orc/-/compare/0.4.40...0.4.41

Also perform some cleanups:

- Dereference `docbook_xsl` alias with `docbook-xsl-nons`.
- Use `finalAttrs` pattern.
- Use `mesonEnable` instead of conditional flags.
- Avoid inheriting from `lib` just for one use each.
- Move `outputs` under `name` and `postPatch` to imperative section.
- Update changelog link to current Git frontend.

jtojnar committed:
gst_all_1.gstreamer: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Format expression
- Use `finalAttrs` pattern
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-plugins-base: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Use `finalAttrs` pattern
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-libav: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Format expression
- Use `finalAttrs` pattern
- Move `outputs` above `src` for consistency with other GStreamer packages
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-plugins-ugly: 1.24.10 → 1.26.0

gst_all_1.gst-libav: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Format expression
- Use `finalAttrs` pattern
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-plugins-good: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
  - darwin dynamic loading fixed https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/7635
  - debug options renamed
- Format expression
- Use `finalAttrs` pattern
- Move environment variable to `env`
- Move `outputs` above `src` for consistency with other GStreamer packages
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-plugins-bad: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Use `finalAttrs` pattern
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-vaapi: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Format expression
- Use `finalAttrs` pattern
- Move `outputs` above `src` for consistency with other GStreamer packages
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-rtsp-server: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Format expression
- Use `finalAttrs` pattern
- Move `outputs` above `src` for consistency with other GStreamer packages
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-plugins-rs: 0.13.3 → 0.13.5

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Format expression
- Do not take `pname` from `finalAttrs`

jtojnar committed:
gst_all_1.gst-devtools: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
  - New tool dots-viewer, requires Rust
- Format expression
- Use `finalAttrs` pattern
- Move `outputs` above `src` for consistency with other GStreamer packages
- Add `updateScript`

jtojnar committed:
gst_all_1.gst-editing-services: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Format expression
- Use `finalAttrs` pattern
- Add `updateScript`

jtojnar committed:
python3.pkgs.gst-python: 1.24.10 → 1.26.0

- Update https://gstreamer.freedesktop.org/releases/1.26/
- Format expression
- Add `updateScript`

jtojnar commented:
Currently failing tests in `gst-plugins-rs`:

```
[1m[32m     Running[0m tests/uriplaylistbin.rs (build/target/x86_64-unknown-linux-gnu/debug/deps/uriplaylistbin-68e02419f63d4dcd)

running 12 tests
test decrease_iterations ... ignored
Fontconfig error: Cannot load default config file: No such file: (null)

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig
test cache ... FAILED
test nb_streams_increasing ... ignored
test missing_file ... ok
test missing_http ... ok
test single_audio ... ok
test single_video ... ok
test multi_audio_video ... ok
test multi_audio ... ok
test infinite_to_finite ... ok
test iterations ... ok
test increase_iterations ... ok

failures:

---- cache stdout ----

thread 'cache' panicked at utils/uriplaylistbin/tests/uriplaylistbin.rs:270:9:
assertion `left == right` failed
  left: 0
 right: 2
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    cache

test result: FAILED. 9 passed; 1 failed; 2 ignored; 0 measured; 0 filtered out; finished in 4.70s

[1m[32m     Running[0m tests/reqwesthttpsrc.rs (build/target/x86_64-unknown-linux-gnu/debug/deps/reqwesthttpsrc-6812e3bc0acc8d2c)

running 16 tests
Fontconfig error: Cannot load default config file: No such file: (null)

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

Fontconfig error: No writable cache directories
	/var/cache/fontconfig
	/homeless-shelter/.cache/fontconfig

thread '<unnamed>' panicked at net/reqwest/tests/reqwesthttpsrc.rs:447:44:
called `Result::unwrap()` on an `Err` value: StateChangeError

thread '<unnamed>' panicked at net/reqwest/tests/reqwesthttpsrc.rs:787:44:
called `Result::unwrap()` on an `Err` value: StateChangeError

thread '<unnamed>' panicked at net/reqwest/tests/reqwesthttpsrc.rs:1150:44:
called `Result::unwrap()` on an `Err` value: StateChangeError

thread '<unnamed>' panicked at net/reqwest/tests/reqwesthttpsrc.rs:390:44:
called `Result::unwrap()` on an `Err` value: StateChangeError

thread '<unnamed>' panicked at net/reqwest/tests/reqwesthttpsrc.rs:715:44:
called `Result::unwrap()` on an `Err` value: StateChangeError
fatal runtime error: failed to initiate panic, error 5
[1m[31merror[0m[1m:[0m test failed, to rerun pass `-p gst-plugin-reqwest --test reqwesthttpsrc`

Caused by:
  process didn't exit successfully: `/build/source/build/target/x86_64-unknown-linux-gnu/debug/deps/reqwesthttpsrc-6812e3bc0acc8d2c` (signal: 6, SIGABRT: process abort signal)



Error: CliError { error: Some(2 targets failed:
    `-p gst-plugin-reqwest --test reqwesthttpsrc`
    `-p gst-plugin-uriplaylistbin --test uriplaylistbin`), exit_code: 101 }

1/1 tests [1;31mFAIL          [0m 1935.01s   exit status 1


Summary of Failures:

1/1 tests [1;31mFAIL          [0m 1935.01s   exit status 1

Ok:                 0   
Expected Fail:      0   
Fail:               1   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   
```

jtojnar committed:
gst_all_1: Disable GLIB_DEBUG

`glib_debug` option (`gobject-cast-checks` prior to 1.26) defaults to `auto`, which would result in the cast checks being disabled on stable versions, if we did not use `-Dauto_features=enabled`. Let’s disable the option as it is meant to be on stable releases: https://gitlab.gnome.org/GNOME/glib/-/blob/main/docs/macros.md#g_enable_debug

Also remove option disabling `glib_asserts`. It was introduced in 800ceaf1bdba85525015c967eb9921977a35090a to remove references to `dev` output caused by `__FILE__` used in assertions but they no longer seem to appear. While the option supported the `auto` means disabled on stable versions feature, the option was actually enabled by default, so the comment was incorrect.

jtojnar reviewed (commented):
- bad and devtools waiting on upstream patch https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/8661

hustlerone commented:
If you don't have the patience to wait for it to be merged you can pull in your PR as a patch itself on top of the commit you're operating

jtojnar committed:
python3.pkgs.gst-python: Fix build with PyGObject 3.52

Still need a second patch for https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/4322

jtojnar commented:
Switched the patches to upstream versions, this should now be good to go.

LordGrimmauld commented:
Hey! This update fails to build:
`sys/nvcodec/meson.build:69:4: ERROR: Problem encountered: The nvcodec was enabled explicitly, but required gstcuda dependency is not found`, please check!

LordGrimmauld commented:
https://gstreamer.freedesktop.org/releases/1.26/
the release notes have some mentions about cuda things and nvcodec. It might however make sense to put that stuff behind the nixpkgs cuda flag and not build with it by default. That'd alleviate the headache of having the patch be potentially breaking for all systems and reduce the risk to only cuda users.

jtojnar commented:
`gst_all_1.gst-plugins-bad` built for me on this branch. What system and commit are you building it on?

LordGrimmauld commented:
> `gst_all_1.gst-plugins-bad` built for me on this branch. What system and commit are you building it on?

i built it on aarch64-linux on latest staging. That does have some commits of my own against gst-bad which is why i first assumed to have broken it myself, but after explicitly reverting those it still failed.

mweinelt commented:
gst-plugins-rs fails in the vendor phase on python-updates

https://hydra.lossy.network/log/xf8gxvpkdzq7rvhx5x7qar40hf0sdp4y-gst-plugins-rs-0.13.5-vendor-staging.drv

jtojnar commented:
The latter is from mopidy, fix in https://github.com/NixOS/nixpkgs/pull/395981

LordGrimmauld commented:
aarch64 build fixed (bodged) in https://github.com/NixOS/nixpkgs/pull/395983

jtojnar commented:
Looking at the source, the only reason why `gstcuda` would not be built on `aarch64-linux` is this `subdir_done` call:

https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/1.26.0/subprojects/gst-plugins-bad/gst-libs/gst/cuda/meson.build#L55

But I do not see why it would regress, it was already present in 1.24:

https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/319f5f07604654b1bf4b58ea416e55324e7d27c4

Howewer, it will be removed in 1.26.1: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/aab14d46e4332046086b0396eac5d450d1437779
